force_torque_sensor_controller/src/force_torque_sensor_controller.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2021, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "force_torque_sensor_controller/force_torque_sensor_controller.hpp"
+
+#include <string>
+#include <vector>
+
+namespace force_torque_sensor_controller
+{
+
+ForceTorqueSensorController::ForceTorqueSensorController()
+: controller_interface::ControllerInterface(),
+  sensor_state_publisher_(nullptr)
+{}
+
+controller_interface::return_type
+ForceTorqueSensorController::init(const std::string & controller_name)
+{
+  auto ret = ControllerInterface::init(controller_name);
+  if (ret != controller_interface::return_type::SUCCESS) {
+    return ret;
+  }
+
+  try {
+    auto node = get_node();
+    node->declare_parameter<std::string>("sensor_name", "");
+    node->declare_parameter<std::vector<std::string>>("state_interface_names", {});
+    node->declare_parameter<std::string>("frame_id", "");
+    node->declare_parameter<int>("fx_range", -1);
+    node->declare_parameter<int>("tz_range", -1);
+  } catch (const std::exception & e) {
+    fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
+    return controller_interface::return_type::ERROR;
+  }
+
+  return controller_interface::return_type::SUCCESS;
+}
+
+CallbackReturn ForceTorqueSensorController::on_configure(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  state_interface_names_ = node_->get_parameter("state_interface_names").as_string_array();
+  if (state_interface_names_.empty()) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'state_interface_names' parameter was empty");
+    return CallbackReturn::ERROR;
+  }
+
+  sensor_name_ = node_->get_parameter("sensor_name").as_string();
```

Comment:
put this first, to keep order: sensor_name, interface_names, frame_id

Reply:
Done!